File: 324/c.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;
#define ull unsigned long long
#define ll long long

#define rep(i,n) for (int i=0; i<(n); i++)

int main() {
    ll N;
    string T;
    cin >> N >> T;

    ll cnt = 0;
    vector <int> ans;

    rep(i,N) {
        string S;
        cin >> S;
        if (S == T) {
            ans.push_back(i+1);
            cnt++;
        }
        else if (T.length() == S.length()+1) {
            if (T.substr(0, S.length()) == S) {
                ans.push_back(i+1);
                cnt++;
            } else {
                rep(j, S.length()) {
                    if (T[j] != S[j]) {
                        if (T.substr(j+1) == S.substr(j)) {
                            ans.push_back(i+1);
                            cnt++;
                        }
                        break;
                    }
                }
            }
        }
        else if (T.length() == S.length()-1) {
            if (T == S.substr(0, T.length())) {
                ans.push_back(i+1);
                cnt++;
            } else {
                rep(j, T.length()) {
                    if (T[j] != S[j]) {
                        if (T.substr(j) == S.substr(j+1)) {
                            ans.push_back(i+1);
                            cnt++;
                        }
                        break;
                    }
                }
            }
        }
        else if (T.length() == S.length()) {
            rep(j, T.length()) {
                if (T[j] != S[j]) {
                    if (T.substr(j+1) == S.substr(j+1)) {
                        ans.push_back(i+1);
                        cnt++;
                    }
                    break;
                }
            }
        }
    }
    cout << cnt << endl;
    rep(k, ans.size()) {
        cout << ans[k] << ' ';
    }
    return 0;
}

File: 324/C.py
N, T = input().split()
N = int(N)

ans = []
cnt = 0

for i in range(N):
    S = input()

    # TがT'と等しいとき
    if T == S:
        cnt += 1
        ans.append(i+1)
    
    # T'がTに一文字挿入されたものであるとき
    elif len(T) == len(S)+1:
        if T[:len(T)-1] == S:
            cnt += 1
            ans.append(i+1)
        else:
            for j in range(len(S)):
                if T[j] != S[j]:
                    if T[j+1:] == S[j:]:
                        cnt += 1
                        ans.append(i+1)
                    break

    # T'がTから一文字削除されたものであるとき
    elif len(T) == len(S)-1:
        if T == S[:len(S)-1]:
            cnt += 1
            ans.append(i+1)
        else:
            for j in range(len(T)):
                if T[j] != S[j]:
                    if T[j:] == S[j+1:]:
                        cnt += 1
                        ans.append(i+1)
                    break
    
    # T'がTの一文字を別の文字に変更したものであるとき
    elif len(T) == len(S):
        for j in range(len(T)):
            if T[j] != S[j]:
                if T[j+1:] == S[j+1:]:
                    cnt += 1
                    ans.append(i+1)
                break

print(cnt)
print(*ans)